homeassistant/components/filter/sensor.py

```diff
@@ -248,6 +250,12 @@ def _update_filter_sensor_state(self, new_state, update_ha=True):
         ):
             self._device_class = new_state.attributes.get(ATTR_DEVICE_CLASS)
 
+        if (
+            self._state_class is None
```

Comment:
Instead of creating a new variable to hold this, you could use `_attr_state_class` directly.

Reply:
renamed 👍